Fix root routing in ember

j-mcnally opened the pull request:
I think there might be a better way to do this, but it works? I'd love feed back I couldn't come up with anything better.

Commit:
Fix root route with ember.

For some reason / was being transformed to /root this aims to work around that. I don't know enough about the integration of everything to make this more elegant.

Commit:
Fix root route with ember.

lancejpollard commented:
Sweet! I'll take a look at this and get to your other posts about it in the morning.

Commit:
Use the user defined action

Use the user defined action for the ember controlActionState

Using r[i] alone was breaking @match routes such as '/team' which would point to a team action when really I wanted the index. This allows me to define it. I have no Idea if this breaks resources.
When i get to point where I am actually using @resources I will loop back around.

lancejpollard commented:
Haha, this code in tower is a mess, thanks for digging in anyway!
